org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend

```diff
@@ -96,14 +96,14 @@ class DocumentSymbolMapper {
 		}
 
 		protected def String getName(QualifiedName qualifiedName) {
-			return qualifiedName?.toString;
+			return qualifiedName?.lastSegment;
```

Comment:
Since the non-hierarchical document symbol service [delegates into the hierarchical one (and flattens the symbols after visiting them in pre-order traversal)](https://github.com/eclipse/xtext-core/blob/e3464552ade1f663065f1ae5bc8ef2040566b1a0/org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolService.xtend#L185) we will show only the last segments for the non-hierarchical case too.

Let me know if this needs more clarification.

Reply:
per se it would be ok for me